frontend/pkg/frontend/routes.go

```diff
@@ -150,6 +152,20 @@ func (f *Frontend) routes(r prometheus.Registerer) *MiddlewareMux {
 		MuxPattern(http.MethodDelete, PatternSubscriptions, PatternResourceGroups, PatternProviders, PatternClusters, PatternNodePools),
 		postMuxMiddleware.HandlerFunc(f.ArmResourceDelete))
 
+	// External Auth
+	mux.Handle(
+		MuxPattern(http.MethodGet, PatternSubscriptions, PatternResourceGroups, PatternProviders, PatternClusters, PatternExternalAuth),
+		postMuxMiddleware.HandlerFunc(f.ArmResourceRead))
```

Comment:
Please move this to the `// Resource read endpoints` group above so it has the appropriate set of middlewares.  In particular, we don't want to lock the subscription for GET requests.